openmdao/devtools/partition_tree_n2.py

```diff
@@ -36,17 +44,32 @@ def _tree_dict(ss, component_execution_orders, component_execution_index):
         children = [_tree_dict(s, component_execution_orders, component_execution_index) for s in ss.subsystems()]
 
         if isinstance(ss, Component):
-            for vname, meta in ss.params.items():
-                dtype=type(meta['val']).__name__
-                children.append(OrderedDict([('name', vname), ('type', 'param'), ('dtype', dtype)]))
-
-            for vname, meta in ss.unknowns.items():
-                dtype=type(meta['val']).__name__
-                implicit = False
-                if meta.get('state'):
-                    implicit = True
-                children.append(OrderedDict([('name', vname), ('type', 'unknown'), ('implicit', implicit), ('dtype', dtype)]))
-
+            if ss.is_active():
+                my_chlist = []
+                for vname, meta in ss.params.items():
+                    dtype=type(meta['val']).__name__
+                    my_chlist.append(OrderedDict([('name', vname), ('type', 'param'), ('dtype', dtype)]))
+
+                for vname, meta in ss.unknowns.items():
+                    dtype=type(meta['val']).__name__
+                    implicit = False
+                    if meta.get('state'):
+                        implicit = True
+                    my_chlist.append(OrderedDict([('name', vname), ('type', 'unknown'), ('implicit', implicit), ('dtype', dtype)]))
+            else:
+                my_chlist = []   # just make an empty list
+
+            if MPI:
+                chlist = system.comm.gather(my_chlist, root=0)
+
+                # now in rank 0, just use the first non-empty entry in the list
+                if system.comm.rank == 0 :
+                    for i, vars_on_rank in enumerate(chlist):
+                        if vars_on_rank:
+                            children.extend(vars_on_rank)
+                            break
```

Comment:
this seems like it will bail out too early to gather all of the local vars from every process.  You may just have to make a set or a dict containing all of them by name so that in the end you'll have them all but no duplicates.